kolibri/plugins/setup_wizard/assets/src/views/onboarding-forms/superuser-credentials-form/index.vue

```diff
@@ -0,0 +1,197 @@
+      adminAccountCreationDescription:
+        'This account allows you to manage your Facility and content on this device.',
+      adminNameFieldLabel: 'Full name',
+      adminUsernameFieldLabel: 'Username',
+      adminPasswordFieldLabel: 'Password',
+      adminPasswordConfirmationFieldLabel: 'Enter password again',
+      // error messages
+      nameFieldEmptyErrorMessage: 'Full name cannot be empty',
+      usernameFieldEmptyErrorMessage: 'Username cannot be empty',
+      usernameCharacterErrorMessage: 'Username can only contain letters, numbers, and underscores',
+      passwordFieldEmptyErrorMessage: 'Password cannot be empty',
+      passwordsMismatchErrorMessage: 'Passwords do not match',
+      facilityFieldEmptyErrorMessage: 'Facility cannot be empty',
+      setupProgressFeedback: 'Setting up your device...',
+    },
+    props: {
+      submitText: {
+        type: String,
+        required: true,
+      },
+    },
+    components: {
+      onboardingForm,
+      kTextbox,
+    },
+    data() {
+      return {
+        name: this.currentName,
+        username: this.currentUsername,
+        password: this.currentPassword,
+        passwordConfirm: this.currentPassword,
+        visitedFields: [],
+      };
+    },
+    computed: {
+      nameErrorMessage() {
+        if (this.name === '') {
+          return this.$tr('nameFieldEmptyErrorMessage');
+        }
+      },
+      usernameErrorMessage() {
+        if (this.username === '') {
+          return this.$tr('usernameFieldEmptyErrorMessage');
+        }
+        if (!/^\w+$/g.test(this.username)) {
+          return this.$tr('usernameCharacterErrorMessage');
+        }
+        return '';
+      },
+      passwordErrorMessage() {
+        if (this.password === '') {
+          return this.$tr('passwordFieldEmptyErrorMessage');
+        }
+        return '';
+      },
+      passwordConfirmErrorMessage() {
+        if (this.passwordConfirm === '') {
+          return this.$tr('passwordFieldEmptyErrorMessage');
+        }
+        if (this.passwordConfirm !== this.password) {
+          return this.$tr('passwordsMismatchErrorMessage');
+        }
+        return '';
+      },
+      nameIsInvalid() {
+        return this.visitedFields.includes('name') && !!this.nameErrorMessage;
+      },
+      usernameIsInvalid() {
+        return this.visitedFields.includes('username') && !!this.usernameErrorMessage;
+      },
+      passwordIsInvalid() {
+        return this.visitedFields.includes('password') && !!this.passwordErrorMessage;
+      },
+      passwordConfirmIsInvalid() {
+        return this.visitedFields.includes('passwordConfirm') && !!this.passwordConfirmErrorMessage;
+      },
+      formIsValid() {
+        return (
+          !this.usernameIsInvalid &&
+          !this.passwordIsInvalid &&
+          !this.passwordConfirmIsInvalid &&
+          !this.facilityIsInvalid
+        );
+      },
+    },
+    methods: {
+      validate(fieldName) {
```

Comment:
Wouldn't this result in repeated field names?

Reply:
It would, but that doesn't affect validation as the `includes` function is supposed to stop at first match. Can work around this, though.

Reply:
Maybe an object? Or just 4 separate variables? :wink: